### DISCUSSION TITLE
winget package

### DISCUSSION THREAD

**Discussion by mtoohey31** [qutebrowser/qutebrowser#6496] [Ideas] [answered]
Hi! I was creating a [winget](https://github.com/microsoft/winget-pkgs) manifest for qutebrowser and I wanted to run it by maintainers of the project before I actually open the PR on the official package repo, here's [my fork](https://github.com/mtoohey31/winget-pkgs) with the manifest.

I've done a decent bit of testing with the manifest locally and it seems to work correctly in all respects except one: when uninstalling the version for `/allusers` with `winget unintsall qutebrowser`, it fails to uninstall, but reports that it uninstalled successfully. I'm not sure what's causing this as there is no output, and it doesn't seem to be a permissions issue as I'm doing the uninstall with the same terminal session as the install. Note that the same failure to uninstall doesn't occur for the package when installing for `/currentuser`. So with regards to this issue, we can either leave it as is, try and figure out why its not working, or remove the `machine` scope installation for `/allusers` and only include the `/currentuser`/`user` scoped version.

The only other thing of note is that Windows SmartScreen is flagging it, but I think that should probably be ok, as they have [a tag](https://github.com/microsoft/winget-pkgs/issues?q=label%3AValidation-SmartScreen) specifically for that kind of issue, so there seems to be a process there.

Let me know what you think!

**Comment by The-Compiler** [comment]
Looks good from a very quick look!

@bitraid could you take a look at the [manifest](https://github.com/mtoohey31/winget-pkgs/blob/d0e8788150fa26368460d1797b2302875847a54c/manifests/q/qutebrowser/qutebrowser/2.2.2/qutebrowser.qutebrowser.installer.yaml) as well as the installer issue @mtoohey31 mentioned, if you find the time?

> The only other thing of note is that Windows SmartScreen is flagging it

Yeah, I guess what might help is signing the installer - see #5020 for a similar issue with macOS. Unfortunately there's quite some effort involved (and also money, though that might be the smaller problem).

**Comment by bitraid** [comment, marked as answer]
The silent uninstall issue comes from the workaround of the Windows elevation bug that existed until Windows 10 version 2004. The upcoming version of the installer has an improved workaround, but for now I'll just open a PR with a quick fix. Also, may I suggest a few changes/additions to the manifest:
```diff
diff --git a/manifests/q/qutebrowser/qutebrowser/2.2.2/qutebrowser.qutebrowser.installer.yaml b/manifests/q/qutebrowser/qutebrowser/2.2.2/qutebrowser.qutebrowser.installer.yaml
index 97e52ec38..b66467b33 100644
--- a/manifests/q/qutebrowser/qutebrowser/2.2.2/qutebrowser.qutebrowser.installer.yaml
+++ b/manifests/q/qutebrowser/qutebrowser/2.2.2/qutebrowser.qutebrowser.installer.yaml
@@ -2,46 +2,55 @@ PackageIdentifier: "qutebrowser.qutebrowser"
 PackageVersion: 2.2.2
 Platform:
   - "Windows.Desktop"
-InstallerType: "exe"
+FileExtensions:
+  - html
+  - htm
+  - url
+Protocols:
+  - http
+  - https
+Commands:
+  - qutebrowser
 InstallModes:
+  - interactive
   - silent
   - silentWithProgress
 Installers:
   - Architecture: x64
-    InstallerType: exe
+    InstallerType: nullsoft
     InstallerUrl: https://github.com/qutebrowser/qutebrowser/releases/download/v2.2.2/qutebrowser-2.2.2-amd64.exe
     InstallerSha256: 65610f77e96573060fb149a46efcfde9bf4fe62fd0d78cd73696a9015afdf521
     InstallerSwitches:
       Custom: /allusers
-      Silent: /S
-      SilentWithProgress: /S
     Scope: machine
+    InstallerLocale: en-US
+    UpgradeBehavior: install
   - Architecture: x64
-    InstallerType: exe
+    InstallerType: nullsoft
     InstallerUrl: https://github.com/qutebrowser/qutebrowser/releases/download/v2.2.2/qutebrowser-2.2.2-amd64.exe
     InstallerSha256: 65610f77e96573060fb149a46efcfde9bf4fe62fd0d78cd73696a9015afdf521
     InstallerSwitches:
       Custom: /currentuser
-      Silent: /S
-      SilentWithProgress: /S
     Scope: user
+    InstallerLocale: en-US
+    UpgradeBehavior: install
   - Architecture: x86
-    InstallerType: exe
+    InstallerType: nullsoft
     InstallerUrl: https://github.com/qutebrowser/qutebrowser/releases/download/v2.2.2/qutebrowser-2.2.2-win32.exe
     InstallerSha256: c909b0a7b7a08376b66ff97d4b1d59d5c205db9c09d022b8fe087db5498b7df1
     InstallerSwitches:
       Custom: /allusers
-      Silent: /S
-      SilentWithProgress: /S
     Scope: machine
+    InstallerLocale: en-US
+    UpgradeBehavior: install
   - Architecture: x86
-    InstallerType: exe
+    InstallerType: nullsoft
     InstallerUrl: https://github.com/qutebrowser/qutebrowser/releases/download/v2.2.2/qutebrowser-2.2.2-win32.exe
     InstallerSha256: c909b0a7b7a08376b66ff97d4b1d59d5c205db9c09d022b8fe087db5498b7df1
     InstallerSwitches:
       Custom: /currentuser
-      Silent: /S
-      SilentWithProgress: /S
     Scope: user
+    InstallerLocale: en-US
+    UpgradeBehavior: install
 ManifestType: installer
 ManifestVersion: 1.0.0
```

**Reply by mtoohey31** [reply]
Awesome, that all looks great! Lmk if you'd like to add that as as PR on my fork, or you just want me to add it in, or if you'd prefer to fork the winget-pkgs repo and create the PR on there yourself, any of the above is good with me!

**Reply by bitraid** [reply]
Just add the changes yourself if you don't mind. Thanks.

**Comment by mtoohey31** [comment]
Ok, so I've added @bitraid's changes to my fork, is there anything else that needs to be done before I open the PR, or should I go ahead and do that?

**Reply by The-Compiler** [reply]
Not from my point of view. Note that I'll release v2.2.3 very soon (probably in the next couple of hours), so you might want to wait until that's released, maybe.

**Reply by mtoohey31** [reply]
Sounds good! I see that v2.2.3 is out now so I'll update things in a little bit when I have a moment and open the PR. Thanks for your help with this!

**Reply by mtoohey31** [reply]
Alright, [it's merged](https://github.com/microsoft/winget-pkgs/pull/15667) so it should show up soon!
